addon/-private/diff-route-info.js

```diff
@@ -0,0 +1,118 @@
+export let paramsDiffer;
+export let createPrefetchChangeSet;
+
+if (gte('3.6.0')) {
+  function createList(enumerable) { // eslint-disable-line no-inner-declarations
+    let out = [];
+
+    if (enumerable === null) return out;
+
+    enumerable.find(item => {
+      out.push(item)
+    });
+    return out;
+  }
+
+  qpsChanged = function _qpsChanged(from, to) {
+    let fromQps = Object.keys(from.queryParams).sort();
+    let toQps = Object.keys(to.queryParams).sort();
+
+    if (fromQps.length !== toQps.length) return true;
+
+    return !toQps.every((qp, i)=> {
+      return fromQps[i] === qp && to.queryParams[qp] === from.queryParams[qp];
+    });
+  }
+
+  shouldRefreshModel = function _shouldRefreshModel(routeQueryParams, infoQueryParams) {
+    let routeQPKeys = Object.keys(routeQueryParams);
+    let infoQPKeys = Object.keys(infoQueryParams);
+    return routeQPKeys.some(key => {
+      return routeQueryParams[key].refreshModel && infoQPKeys.indexOf(key) > -1;
+    });
+  }
+
+  function paramsMatch(from, to) { // eslint-disable-line no-inner-declarations
+    return to.paramNames.every((paramName, i) => {
+      return from.paramNames[i] === paramName && from.params[paramName] === to.params[paramName];
+    });
+  }
+
+  pathsDiffer = function _pathsDiffer(from, to) {
+    return !to.every((info, i) => {
+      return info.name === from[i].name;
+    })
+  }
+
+
+  paramsDiffer = function _paramsDiffer(from, to) {
+    return !to.every((info, i) => {
+      let _from = from[i];
+      let _to = info;
+      return _to.paramNames.length === _from.paramNames.length && paramsMatch(_from, _to)
+    });
+  }
+
+  function qpsDiffer(privateRouter, to, transition) { // eslint-disable-line no-inner-declarations
+    let routes = getPrefetched(privateRouter, to);
+    if (transition.from === null) {
+      return { shouldCall: true, for: routes };
+    }
+
+    if (qpsChanged(transition.from, transition.to)) {
+      let prefetchRoutes = [];
+
+      routes.forEach((info) => {
+        let { route } = info;
+        if (shouldRefreshModel(route.queryParams, transition.to.queryParams)) {
+          prefetchRoutes.push(info);
+        }
+      });
+
+      return { shouldCall: true, for: prefetchRoutes };
+    }
+
+    return { shouldCall: false, for: [] };
+  }
+
+  function getPrefetched(privateRouter, to) { // eslint-disable-line no-inner-declarations
+    let routes = [];
+    for (let i = 0; i < to.length; i++) {
+      let info = to[i];
+      let route = privateRouter.getRoute(info.name);
+      if (route !== undefined && route !== null) {
+        routes.push({ route, fullParms: assign({}, info.params, { queryParams: info.queryParams }) });
+      }
+    }
+
+    return routes;
+  }
+
+  createPrefetchChangeSet = function _createPrefetchChangeSet(privateRouter, transition) {
+    let toList = createList(transition.to);
+    let fromList = createList(transition.from);
+
+    // Paths Changed
+    if (toList.length > fromList.length) {
```

Comment:
I'm curious why `>` was opted for instead of `!==` here.

Reply:
Good catch.